File: TheLoversApi/src/main/java/com/strilicherk/theloversapi/feature_user/domain/mappers/UserMapper.java
package com.strilicherk.theloversapi.feature_user.domain.mappers;

import com.strilicherk.theloversapi.core.domain.model.user_location.UserLocation;
import com.strilicherk.theloversapi.core.domain.model.user_location.UserLocationRequestDTO;
import com.strilicherk.theloversapi.feature_user.domain.dto.UserResponseDTO;
import com.strilicherk.theloversapi.core.domain.model.user.User;

public class UserMapper {

    public static UserResponseDTO fromUserToUserResponseDto(User user) {
        return new UserResponseDTO(
                user.getId(),
                user.getName(),
                user.getEmail(),
                user.getPhone(),
                user.getUserVerified(),
                user.getBirthDate(),
                user.getGender() != null ? user.getGender().getId() : null,
                toLocationRequestDTO(user.getLocation()),
                user.getRole().getId()
        );
    }

//    public static User

//    public static User fromUserUpdateDtoToUser(User user, UserUpdateDTO dto) {
//        return new User(
//            if (dto.email() != null) user.setEmail(dto.email());
//            if (dto.phone() != null) user.setPhone(dto.phone());
//        )
//    }
//
//    public static void applyUpdateDTO(User user, UserUpdateDTO dto) {
//        if (dto.name() != null) user.setName(dto.name());
//
//        if (dto.birthDate() != null) user.setBirthDate(dto.birthDate());
//    }

    //    public static User fromCreateDTO(UserCreateDTO dto) {
//        User user = new User();
//        user.setPhone(dto.phone());
//        user.setUserVerified(false);
//        return user;
//    }
//
//    public static void applyUpdateDTO(User user, UserUpdateDTO dto) {
//        if (dto.name() != null) user.setName(dto.name());
//        if (dto.email() != null) user.setEmail(dto.email());
//        if (dto.phone() != null) user.setPhone(dto.phone());
//        if (dto.birthDate() != null) user.setBirthDate(dto.birthDate());
//        // genderId e locationId exigem fetch da entidade, feito no service
//    }
//
    public static UserLocationRequestDTO toLocationRequestDTO(UserLocation location) {
        if (location == null) return null;
        return new UserLocationRequestDTO(
                location.getUser().getId(),
                location.getId(),
                location.getCountry(),
                location.getState(),
                location.getCity()
//                location.getLatitude(),
//                location.getLongitude()
        );
    }
}

File: TheLoversMobileWeb/composeApp/src/commonMain/kotlin/org/example/thelovers/feature_user/data/mapper/UserMapper.kt
package org.example.thelovers.feature_user.data.mapper

import kotlinx.datetime.LocalDate
import kotlinx.datetime.LocalDateTime
import org.example.thelovers.feature_user.data.dto.UserResponseDTO
import org.example.thelovers.feature_user.domain.model.User
import org.example.thelovers.feature_user.domain.model.UserEntity

object UserMapper{
    fun UserEntity.toUserModel(): User {
        return User(
            id = id,
            name = name,
            email = email,
            phone = phone,
            birthDate = birthDate?.let { LocalDate.parse(it) },
            genderId = genderId,
            roleId = roleId,
            locationId = locationId,
            userVerified = userVerified,
            createdAt = LocalDateTime.parse(createdAt),
            updatedAt = LocalDateTime.parse(updatedAt)
        )
    }

    fun User.toEntityModel(): UserEntity {
        return UserEntity(
            id = id,
            name = name,
            email = email,
            phone = phone,
            birthDate = birthDate?.toString(),
            genderId = genderId,
            roleId = roleId,
            locationId = locationId,
            userVerified = userVerified,
            createdAt = createdAt.toString(),
            updatedAt = updatedAt.toString()
        )
    }

    fun UserResponseDTO.toUserModel(): User {
        return User(
            id = id,
            name = name,
            email = email,
            phone = phone,
            userVerified = userVerified,
            birthDate = birthDate.let { if (it.isNullOrEmpty()) null else LocalDate.parse(it) },
            genderId = genderId,
            locationId = userLocation?.id,
            roleId = roleId,
            createdAt = createdAt.let { LocalDateTime.parse(it) },
            updatedAt = updatedAt.let { LocalDateTime.parse(it) },
        )
    }

    fun UserResponseDTO.toUserEntity(): UserEntity {
        return UserEntity(
            id = id,
            name = name,
            email = email,
            phone = phone,
            userVerified = userVerified,
            birthDate = birthDate,
            genderId = genderId,
            locationId = userLocation?.id,
            roleId = roleId,
            createdAt = createdAt,
            updatedAt = updatedAt,
        )
    }
}